packages/oats-runtime/src/make.ts

```diff
@@ -457,6 +457,10 @@ export function makeObject<
       result[index] = propResult.success();
     }
     for (const index of Object.keys(value)) {
+      // do not consider props that have undefined value
```

Comment:
do we want to prevent props with undefined values when additionalProperties: false?

Reply:
mmm does it matter if we have additionalProperties true or false? should this be done consistently regardless? 

from the developer point of view, I suppose if additionalProperties is true then we can leave them be, WDYT?

Reply:
additionalProperties: false means that no additional properties are allowed. Might be that we do not consider undefined valued props as additional and are ok to ignore and drop those.

Reply:
good point though: it theoretically might be a breaking (and possibly tricky to spot) change if additionalProperties: true is set and we would start omitting the props with undefined values. Somebody could be constructing values using `.make({ prop: undefined})` and dropping `prop` would change the resulting value. Not an issue when json serializing but an issue for `Object.keys(theMadeValue)`.